ruby/example_code/s3/tests/test_auth_request_object_keys.rb

```diff
@@ -0,0 +1,20 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX - License - Identifier: Apache - 2.0
+
+require_relative '../auth_request_object_keys.rb'
+
+describe '#can_list_bucket_objects?' do
+  let(:s3) { Aws::S3::Client.new(stub_responses: true) }
+  let(:bucket_name) { 'my-bucket' }
+
+  it "lists the objects' keys in the specified bucket" do
+    objects_data = s3.stub_data(:list_objects_v2,
+      contents: [
+        { key: 'my-file-1.txt' },
+        { key: 'my-file-2.txt' }
+      ]
+    )
+    s3.stub_responses(:list_objects_v2, objects_data)
+    expect(can_list_bucket_objects?(s3, bucket_name)).to be
```

Comment:
expect these to be true

`expect(can_list_bucket_objects?(s3, bucket_name)).to be true`

Reply:
> Is a Class a requirement here? 

No, classes are not a requirement here. In fact, I feel like the prior rather heavy reliance on custom classes seemed to distract both visually and conceptually from what we're trying to achieve here for users: brevity, simplicity, and ease of reuse.

There are certainly great reasons to use custom classes; I just feel that in these particular code examples it's too complex.

Reply:
Updated .to be to .to be true in unit test per your proposal.